garmin_device_xml.cc

```diff
@@ -127,10 +128,7 @@ const gdx_info*
 gdx_read(const char* fname)
 {
   // Test file open-able before gb_open gets a chance to fatal().
-  FILE* fin = ufopen(QString::fromUtf8(fname), "r");
-
-  if (fin) {
-    fclose(fin);
+  if (QFile::exists(fname) && QFile(fname).open(QIODevice::ReadOnly)) {
```

Comment:
I realized QFile::exists() by itself isn't strictly enough, since just because a file exists (and can be seen) doesn't mean it can be opened. So I added a QFile::open() call as well. Come to think of it, I suppose that makes the QFile::exists() call redundant, assuming QFile::open() will fail "softly" without generating an exception/abort/nasty output.